src/backend/plugins/config-gnome/config-gnome.c

```diff
@@ -0,0 +1,245 @@
+  GObject parent_instance;
+  GSettings *proxy_settings;
+  GSettings *http_proxy_settings;
+  GSettings *https_proxy_settings;
+  GSettings *ftp_proxy_settings;
+  GSettings *socks_proxy_settings;
+  gboolean settings_found;
+};
+
+typedef enum {
+  GNOME_PROXY_MODE_NONE,
+  GNOME_PROXY_MODE_MANUAL,
+  GNOME_PROXY_MODE_AUTO
+} GnomeProxyMode;
+
+static void px_config_iface_init (PxConfigInterface *iface);
+void peas_register_types (PeasObjectModule *module);
+
+G_DEFINE_FINAL_TYPE_WITH_CODE (PxConfigGnome,
+                               px_config_gnome,
+                               G_TYPE_OBJECT,
+                               G_IMPLEMENT_INTERFACE (PX_TYPE_CONFIG, px_config_iface_init))
+
+enum {
+  PROP_0,
+  PROP_CONFIG_OPTION
+};
+
+static void
+px_config_gnome_init (PxConfigGnome *self)
+{
+  GSettingsSchemaSource *source = g_settings_schema_source_get_default ();
+  GSettingsSchema *proxy_schema;
+
+  if (!source)
+    return;
+
+  proxy_schema = g_settings_schema_source_lookup (source, "org.gnome.system.proxy", TRUE);
+
+  self->settings_found = proxy_schema != NULL;
+  g_clear_pointer (&proxy_schema, g_settings_schema_unref);
+
+  if (!self->settings_found)
+    return;
+
+  self->proxy_settings = g_settings_new ("org.gnome.system.proxy");
+  self->http_proxy_settings = g_settings_new ("org.gnome.system.proxy.http");
+  self->https_proxy_settings = g_settings_new ("org.gnome.system.proxy.https");
+  self->ftp_proxy_settings = g_settings_new ("org.gnome.system.proxy.ftp");
+  self->socks_proxy_settings = g_settings_new ("org.gnome.system.proxy.socks");
+}
+
+static void
+px_config_gnome_set_property (GObject      *object,
+                              guint         prop_id,
+                              const GValue *value,
+                              GParamSpec   *pspec)
+{
+  switch (prop_id) {
+    case PROP_CONFIG_OPTION:
+      break;
+
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+      break;
+  }
+}
+
+static void
+px_config_gnome_get_property (GObject    *object,
+                              guint       prop_id,
+                              GValue     *value,
+                              GParamSpec *pspec)
+{
+  switch (prop_id) {
+    case PROP_CONFIG_OPTION:
+      break;
+
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+      break;
+  }
+}
+
+static void
+px_config_gnome_class_init (PxConfigGnomeClass *klass)
+{
+  GObjectClass *object_class = G_OBJECT_CLASS (klass);
+
+  object_class->set_property = px_config_gnome_set_property;
+  object_class->get_property = px_config_gnome_get_property;
+
+  g_object_class_override_property (object_class, PROP_CONFIG_OPTION, "config-option");
+}
+
+static gboolean
+px_config_gnome_is_available (PxConfig *config)
+{
+  PxConfigGnome *self = PX_CONFIG_GNOME (config);
+  const char *desktops;
+
+  if (!self->settings_found)
+    return FALSE;
+
+  desktops = getenv ("XDG_CURRENT_DESKTOP");
```

Comment:
IMO it should check for `XDG_CURRENT_DESKTOP` before doing any initialization

Reply:
Reworked GNOME availability check.